### PR TITLE
Update to use shared renovate config

### PR DESCRIPTION
**Pull request by dprince**

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -1,47 +1,5 @@
...
-    },
-    {
-      "groupName": "sigs.k8s.io/controller-runtime",
-      "matchPackagePatterns": ["^sigs.k8s.io/controller-runtime"],
-      "allowedVersions": "< 0.15.0",
-      "enabled": true
-    }
-  ],
-  "postUpgradeTasks": {
-    "commands": ["make gowork", "make tidy", "make manifests generate"],
```
Comment (gibizer) [inline renovate.json:43]: It seems the common config regresses this as it only runs make tidy https://github.com/openstack-k8s-operators/renovate-config/blob/79f5b0ba87339a43b8c7bb9f9391aa3f7152fab1/default.json#L47

Comment (dprince) [reply renovate.json:43]: okay, so I guess we can drop postUpgrade from common and then just have it in each operator. Probably safer that way as it depends on the Makefile too right

Comment (dprince) [reply renovate.json:43]: https://github.com/openstack-k8s-operators/renovate-config/pull/1

Comment (stuggi) [reply renovate.json:43]: +1 sounds good to me

Comment (gibizer) [reply renovate.json:43]: I'm OK with that as I can imagine some repository does not have manifests and generate targets.

File: renovate.json [hunk 2]
```diff
@@ -1,43 +1,6 @@
 {
-"extends": [
-    "config:base"
-  ],
-  "dependencyDashboard": true,
-  "logFileLevel": "trace",
-  "enabledManagers": ["gomod"],
-  "postUpdateOptions": ["gomodTidy"],
-  "constraints": {
-    "go": "1.19"
```
[the remark this answers is not in the document]
Comment (dprince) [reply renovate.json:10]: Good call. Lets keep go in this file too.

**Comment by stuggi** [comment]
+1 from me

**Review by gibizer — COMMENTED** [review]
looks good to me too

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**